### DISCUSSION TITLE
[Q] Support for ECMAScript module

### DISCUSSION THREAD

**Discussion by mdantonio** [exceljs/exceljs#2404] [Q&A]
## 💬 Questions and Help

Hello, I'm opening this issue because I did not find other discussions about that, please redirect me on the proper issue if you are already discussing it. I was looking for any solution for the Angular 10 warnings (_CommonJS or AMD dependencies can cause optimization bailouts._), rather that simply ignore them by configuring allowedCommonJsDependencies in angular.json.
I realized that a ECMAScript module would be needed to make Angular happy. Is there any plan to build exceljs in that way?

Thank you a lot for the support

**Comment by yume-chan** [comment]
Since the source code is using cjs module format, I think it must be merged to esm first so it can be built that way.

**Comment by hmendezm** [comment]
Hi everybody,

guys thank you very much for this library. I upgraded Angular to 13 and I get the "'exceljs'. CommonJS or AMD..." warning. Do you have news about this? is there a plan to move this library to ECMAScript module.

Best
Hmendez

**Comment by jakehockey10** [comment]
This would be super neat

**Comment by Catscratch** [comment]
+1 
A conversion to esm would be nice.

**Comment by Kombuchelada** [comment]
I imagine it could be difficult for such an extensive library to get this conversion, but with ESM being the new de-facto standard, I think it'd be super neat. It'd also get rid of my console warnings from Angular :)

Great work on the library y'all. This is an indispensable tool.

**Comment by rogeriopgp** [comment]
Hello!

    Any news about this issue? I have the same problem here: big js file because of exceljs.

**Comment by Fatichti** [comment]
Hello,

No news about it ?

**Comment by softboy99** [comment]
does this lib still alive?

**Comment by gameroman** [comment]
This package is not very maintained, I'd recommend trying out one of the more actively maintained forks

https://github.com/excel-js/exceljs

https://github.com/cjnoname/excelts

https://github.com/protobi/exceljs

You can also see a [comparison on npmx](https://npmx.dev/compare?packages=exceljs,@cj-tech-master/excelts,@excel.js/exceljs,@protobi/exceljs)
